src/Microsoft.ML.Core/Data/IEstimator.cs

```diff
@@ -146,17 +146,28 @@ public SchemaShape(IEnumerable<Column> columns)
             out ColumnType itemType,
             out bool isKey)
         {
-            if (type.IsKnownSizeVector)
-                vecKind = Column.VectorKind.Vector;
-            else if (type.IsVector)
-                vecKind = Column.VectorKind.VariableVector;
+            if (type is VectorType vectorType)
```

Comment:
>type is VectorType vectorType [](start = 16, length = 29)

did you touch this code before you wrote extensions, or what's the reason for not using them?